libraries/shared/transformer/seed_node_transformer.go

```diff
@@ -0,0 +1,31 @@
+// VulcanizeDB
+// Copyright © 2019 Vulcanize
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package transformer
+
+import (
+	"github.com/vulcanize/vulcanizedb/pkg/config"
+	"github.com/vulcanize/vulcanizedb/pkg/core"
+	"github.com/vulcanize/vulcanizedb/pkg/datastore/postgres"
+)
+
+type SeedNodeTransformer interface {
+	Init() error
+	Execute() error
+	GetConfig() config.Subscription
+}
+
+type SeedNodeTransformerInitializer func(db *postgres.DB, subCon config.Subscription, client core.RpcClient) SeedNodeTransformer
```

Comment:
Could you explain when a `SeedNodeTransformer` would need to be used? Not sure I understand the relationship between this and an EventTransformer for example.

Reply:
This is just the general interface for writing transformers which utilize the seed node, analogous to how the `EventTransformer` interface is for writing transformers which utilize direct fetching of events from geth. It's not broken up into `storage`, `event`, etc subtypes since we can get all types of data from the seed node in the same payload/subscription. Example of transformer that fits this interface is https://github.com/vulcanize/account_transformers/blob/blockscout/transformers/account/seed/transformer.go

I'd like to discuss this in more detail maybe at our next meeting, as I'm not entirely sure writing seed node based transformers as plugins like we are writing the other ones makes a whole lot of sense (since seed node is interacted with over rpc).

Reply:
Can we have this discussion here or in a chat room instead of on a call? Thanks.

Reply:
Since seed node transformers interact with the seed node over rpc as opposed to how current transformers are run as part of the same process we no longer need to load the transformers as a plugin to vdb in order to link the vdb binary to a set of transformers. We still want to be able to compose a set of arbitrary transformers to interact with vdb over rpc, but this can be done without plugins. Basically we can have multiple binaries now- the VulcanizeDB seed node binary which is completely static and then build separate binaries for sets of transformers.

Reply:
Wouldn't the transformers talking to each other over RPC?

Reply:
That's not how transformers have been written previously but that's an option. That would mean each transformer is its own process which will come with some additional overhead (having to relay data and syncrhonize processing between multiple separate processes).